enter.c

```diff
@@ -225,13 +226,65 @@ int enter(struct entry_settings *opts)
 		}
 	}
 
+	int rootpid = getpid();
+	bool req_cgroup_clean = false;
+	int cleanfd;
+
+	// Do not join the root cgroup if limits are to be applied, otherwise bst needs to create sub-hierarchies
+	if (!opts->nactiveclimits && opts->cgroup.present) {
+		int cgroupfd = open(opts->cgroup.cgroup_path, O_PATH | O_DIRECTORY);
+		if (cgroupfd == -1) {
+			err(1, "unable open cgroup %s", opts->cgroup.cgroup_path);
+		}
+
+		join_cgroup(opts->cgroup.cgroup_path, cgroupfd);
+
+		close(cgroupfd);
+
+	} else if (opts->nactiveclimits && opts->cgroup.present) {
+
+		int rootfd = open(root, O_PATH, 0);
+		if (rootfd == -1) {
+			 err(1, "open %s", root);
+		}
+
+		const char * rootcgroup = opts->cgroup.cgroup_path;
+		const char * subcgroup = makepath("%s/bst.%d", rootcgroup,rootpid);
```

Comment:
micro-nit: missing space after `rootcgroup,`.

I should probably look into putting some uncrustify config for code formatting.